parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -201,6 +211,69 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return internalReader.nextKeyValue();
   }
 
+  /**
+   * Reads the next batch of rows. This method is used for reading complex types
+   * or arbitrary objects and calls the converters eventually to materialize the record.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @param clazz the class of the record type that will be filled into the column vector
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+  @Override
+  public RowBatch nextBatch(RowBatch previous, Class<T> clazz) throws IOException {
+    throw new UnsupportedOperationException("Complex types is not supported now");
+  }
+
+  /**
+   * Reads the next batch of rows. This method is used for reading primitive types
+   * and does not call the converters at all.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+  @Override
+  public RowBatch nextBatch(RowBatch previous) throws IOException {
+    MessageType requestedSchema = readContext.getRequestedSchema();
+    List<ColumnDescriptor> columns = requestedSchema.getColumns();
+    int nColumns = columns.size();
+    ColumnVector[] columnVectors;
+
+    RowBatch rowBatch = previous;
+    if (rowBatch == null) {
+      rowBatch = new RowBatch();
+    }
+
+    if (rowBatch.getColumns() == null) {
+      columnVectors = new ColumnVector[nColumns];
+      rowBatch.setColumns(columnVectors);
+    } else {
+      columnVectors = rowBatch.getColumns();
+    }
+
+    MessageType[] columnSchemas = new MessageType[nColumns];
+    for (int i = 0; i < nColumns; i++) {
+      ColumnVector columnVector = columnVectors[i];
+      columnSchemas[i] = new MessageType(requestedSchema.getFieldName(i), requestedSchema.getType(i));
+
+      if (columnVector == null) {
+        columnVector = ColumnVector.from(columns.get(i));
+      }
+
+      rowBatch.getColumns()[i] = columnVector;
+    }
+
+    boolean hasMoreRecords = readVectors(rowBatch.getColumns(), columnSchemas);
+    return hasMoreRecords ? rowBatch : null;
+  }
+
+  private boolean readVectors(ColumnVector[] vectors, MessageType[] columns) throws IOException {
+    try {
+      return internalReader.nextBatch(vectors, columns);
+    } catch (InterruptedException e) {
```

Comment:
I will remove `InterruptedException` from the signature of `nextBatch()` so you should update this eventually.